Controllers/Repository.php

```diff
@@ -0,0 +1,136 @@
+ */
+
+use Shopware\Models\Category\Category;
+
+class Shopware_Plugins_Frontend_NostoTagging_Controllers_Repository
+{
+    /**
+     * Returns an array of articles id's that are active
+     * and has the same category id of the given shopware category
+     *
+     * @param \Shopware\Models\Category\Category $category
+     * @param $pageSize
+     * @param $currentOffset
+     * @param $id
+     * @return array
+     */
+    public function getActiveArticlesIdsByCategory(
+        Category $category,
+        $pageSize,
+        $currentOffset,
+        $id
+    ) {
+        $builder = Shopware()->Models()->createQueryBuilder();
+        $result = $builder->select('articles.id')
+            ->from('\Shopware\Models\Article\Article', 'articles')
+            ->innerJoin(
+                '\Shopware\Models\Article\Detail',
+                'details',
+                \Doctrine\ORM\Query\Expr\Join::WITH,
+                'articles.mainDetailId = details.id'
+            )
+            ->innerJoin(
+                'articles.categories',
+                'c'
+            )
+            ->where('articles.active = 1')
+            ->andWhere('c.path LIKE :path')
+            // Since the path in the database is saved with || between
+            // the parents ids, we concatenate those and get all child
+            // categories from the given language.
+            ->setParameter('path', '%|'.(int)$category->getId().'|%');
+        if (!empty($id)) {
+            $result = $result->andWhere('details.number = :id')
+                ->setParameter('id', $id)
+                ->getQuery();
+        } else {
+            $result = $result->orderBy('articles.added', 'DESC')
+                ->setFirstResult($currentOffset)
+                ->setMaxResults($pageSize)
+                ->getQuery();
+        }
+        return $result->getResult(\Doctrine\ORM\AbstractQuery::HYDRATE_ARRAY);
+    }
+
+    /**
+     * @param \Shopware\CustomModels\Nosto\Account\Account $account
+     * @return bool
+     */
+    public function isAccountAlreadyRegistered(
+        \Shopware\CustomModels\Nosto\Account\Account $account
+    ) {
+        /** @var Shopware\CustomModels\Nosto\Account\Account $existingAccount */
+        $existingAccount = Shopware()
+            ->Models()
+            ->getRepository("Shopware\CustomModels\Nosto\Account\Account")
+            ->findOneBy(array('name' => $account->getName()));
+
+        // If an account has been found, and the shop id is different from current shop, then it means
+        // the admin is trying to map same nosto account to two sub shops. It is not allowed.
+        if ($existingAccount != null && $existingAccount->getShopId() !== $account->getShopId()) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param $pageSize
+     * @param $currentOffset
+     * @param $id
+     * @return array
+     */
+    public function getCompletedOrders($pageSize, $currentOffset, $id)
```

Comment:
This should not be in this repo. Create own repo for fetching orders

Reply:
Fixed